R/qsys_sge.r

```diff
@@ -13,14 +13,20 @@ SGE = R6::R6Class("SGE",
 
         submit_jobs = function(...) {
             opts = private$fill_options(...)
-            private$job_id = opts$job_name
+            private$job_name = opts$job_name
             filled = private$fill_template(opts)
 
-            success = system("qsub", input=filled, ignore.stdout=TRUE)
-            if (success != 0) {
+            private$qsub_stdout  = system2("qsub", input=filled, stdout = T)
```

Comment:
Why store this in a class variable? If your `set_job_id` takes `qsub_stdout` we save a class-level variable.

Reply:
I chose to store this in a class variable to make overriding `set_job_id` more straightforward between implementations, without having to reimpliment the more complex `submit_jobs` method. 

`set_job_id` takes no arguments, and gets its value from a private variable that depends on implementation.

Reply:
I'll change everything else now, I won't change this just yet unless you insist

Reply:
> without having to reimpliment the more complex `submit_jobs` method.

This is not required either way. Why not just make the `set_job_id` method take `qsub_stdout` as argument?

Reply:
I will do as you ask, and for SGE, I will make it ignore its input argument and use `job_name`.